### PR TITLE
utils/github: paginate pull request commits API

### PR DESCRIPTION
**Pull request by jonchang**
This is needed to run `brew pr-pull` on jumbo pull requests like https://github.com/Homebrew/homebrew-core/pull/62560.

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [ ] Have you successfully run `brew man` locally and committed any changes?

-----

### CODE REVIEW

File: Library/Homebrew/utils/github.rb [hunk 1]
```diff
@@ -759,7 +759,24 @@ def create_bump_pr(info, args:)
...
+    commits_api = pr_data["commits_url"]
+    commit_count = pr_data["commits"]
+    page = 0
+    commits = []
+
+    loop do
+      page += 1
+      result = open_api(commits_api + "?per_page=#{per_page}&page=#{page}")
+      commits.concat(result.map { |c| c["sha"] })
+      break if result.empty? || page * per_page >= commit_count
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/github.rb:773]: The potential for this to be infinite concerns me. Could we set a maximum upper bound here?

**Review by MikeMcQuaid — APPROVED** [review]
One question otherwise 👍🏻

File: Library/Homebrew/utils/github.rb (hunk 1 above)
Comment (jonchang) [reply Library/Homebrew/utils/github.rb:773]: Well, the API request limit will kick in at 49 pages and blow up the function call anyway. Is that an acceptable upper bound?

Comment (MikeMcQuaid) [reply Library/Homebrew/utils/github.rb:773]: Yeh! Could that be hardcoded on our end, too, so we're not reliant on GitHub doing what they say they will 😅

Comment (jonchang) [reply Library/Homebrew/utils/github.rb:773]: Ok, how about raising an error if we hit 50 pages, or the number of commits to be pulled is greater than 5000

Comment (MikeMcQuaid) [reply Library/Homebrew/utils/github.rb:773]: Sounds good 👍🏻 A numbered loop with an early `break` may make sense over a `loop` in that case.

**Comment by jonchang** [comment]
Ok, this is updated.

**Review by MikeMcQuaid — APPROVED** [review]
Thanks for the changes, looks great.

**Pull request merged**